packages/react-instantsearch/src/core/highlight.js

```diff
@@ -1,67 +1,75 @@
-import React from 'react';
-
-function extractTagName(highlightedValue) {
-  const tagParse = (/<(.+?)>/).exec(highlightedValue);
-  return tagParse === null ? null : tagParse[1];
-}
+import get from 'lodash/get';
 
 /**
- * This function transforms an highlighted response
- * to an array of React elements. This should be
- * used in place of dangerouslySetInnerHTML when
- * using the highlightReasult provided by Algolia.
- *
- * The function autodetects which tag is used in
- * results to highlight, and creates the corresponding
- * React element.
+ * Find an highlighted attribute give a path `pathToAttribute`, parses it,
+ * and provided an array of objects with the string value and a boolean if this
+ * value is highlighted.
  *
- * It also creates keys on React elements to avoid warnings.
+ * In order to use this feature, highlight must be activated in the configruation of
+ * the index. The `preTag` and `postTag` attributes are respectively highlightPreTag and
+ * highligtPostTag in Algolia configuration.
  *
- * The genegrated highlighted Elements use the `ais-hightlightedValue`
- * class name attached.
- *
- * @param {string} attribute the name of the attribute for which we want the highlighted value
- * @param {object} hit the hit from the algolia response
- * @returns {Array.<string|ReactElement>} the array of react element
- * @example
- * const ProductHits = connectHits(({hits}) => {
- *   const hitComponents = hits.map( hit =>
- *     <div key={hit.objectID}>
- *       <span className="hit-name">{hightlight('name', hit)}</span>
- *     </div>
- *   );
+ * @param {string} preTag - string used to identify the start of an highlighted value
+ * @param {string} postTag - string used to identify the end of an highlighted value
+ * @param {string} pathToAttribute - path to the highlighted attribute in the results
+ * @param {object} hit - the actual hit returned by Algolia.
+ * @return {object[]} - An array of {value: string, isDefined: boolean}.
+ */
+export default function parseAlgoliaHit({
```

Comment:
If file is highlight.js we should rename export to `export default function highlight`. We do this in other modules I believe.